service-workers/service-worker/resources/fetch-event-async-respond-with-worker.js

```diff
@@ -1,19 +1,18 @@
-var result;
-
-self.addEventListener('message', function(event) {
-    event.data.port.postMessage(result);
-  });
+importScripts('./service-worker-recorder.js');
 
 self.addEventListener('fetch', function(event) {
     setTimeout(function() {
         try {
           event.respondWith(new Response());
-          result = 'FAIL: did not throw';
+          ServiceWorkerRecorder.worker.save('FAIL: did not throw');
```

Comment:
Definitely. This was an issue in multiple places, but correcting it in this
file in particular required a slightly more involved change.